### PR TITLE
wallet: allow to open in read-only mode

### PR DESCRIPTION
**Pull request by fyrchik**
If wallet was opened via `NewWalletFromFile`, open it as read-only first
and re-open for write if needed. Store write flag in a special field.
Another solution would be to check flags via `fcntl` but it is less portable.

Signed-off-by: Evgeniy Stratonikov <evgeniy@nspcc.ru>

### CODE REVIEW

File: pkg/wallet/wallet.go [hunk 1]
```diff
@@ -56,7 +59,7 @@ func NewWallet(location string) (*Wallet, error) {
 
 // NewWalletFromFile creates a Wallet from the given wallet file path.
 func NewWalletFromFile(path string) (*Wallet, error) {
-	file, err := os.OpenFile(path, os.O_RDWR, os.ModeAppend)
+	file, err := os.Open(path)
```
Comment (roman-khimov) [inline pkg/wallet/wallet.go:62]: Why do we keep this handle in `Wallet` at all? We read whole file, we don't really care about it afterwards until there is a need to update it (in `writeRaw`).

File: pkg/wallet/wallet.go [hunk 2]
```diff
@@ -56,12 +56,13 @@ func NewWallet(location string) (*Wallet, error) {
...
 	if err != nil {
 		return nil, err
 	}
+	defer file.Close()
+
 	wall := &Wallet{
-		rw:   file,
```
Comment (roman-khimov) [inline pkg/wallet/wallet.go:64]: Do we need `rw` field? Read/close, write/close?

File: pkg/wallet/wallet.go [hunk 3]
```diff
@@ -194,10 +170,8 @@ func (w *Wallet) JSON() ([]byte, error) {
 }
 
 // Close closes the internal rw if its an io.ReadCloser.
+// Deprecated: will be removed in future versions.
```
Comment (roman-khimov) [inline pkg/wallet/wallet.go:173]: Not sure about this. We can and should scrub private key memory here.

Comment (roman-khimov) [reply pkg/wallet/wallet.go:173]: @fyrchik? It might get deprecated, but I don't think it is now.

Comment (fyrchik) [reply pkg/wallet/wallet.go:173]: Fixed!

Comment (roman-khimov) [reply pkg/wallet/wallet.go:173]: Nothing changed. Forgot to push?

Comment (fyrchik) [reply pkg/wallet/wallet.go]: Yes, fixed.

**Pull request merged**